flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/stream/sql/SqlITCase.scala

```diff
@@ -411,6 +411,128 @@ class SqlITCase extends StreamingWithStateTestBase {
     assertEquals(expected.sorted, StreamITCase.testResults.sorted)
   }
 
+  @Test
+  def testBoundPartitionedEventTimeWindowWithRange(): Unit = {
+    val data = Seq(
```

Comment:
I think a bit more diverse test data would be good to cover more corner cases. 
Basically check that each loop is correctly triggered (retract data of two (or more timestamps), emit different rows in one timestamp, etc.